plugins/CoreAdminHome/OptOutManager.php

```diff
@@ -111,6 +114,50 @@ public function setTitle($title)
     }
 
     /**
+     * @param string $key
+     * @param string $value
+     * @param bool $override
+     *
+     * @return bool
+     */
+    public function addQueryParameter($key, $value, $override = true)
+    {
+        if (!isset($this->queryParameters[$key]) || true === $override) {
+            $this->queryParameters[$key] = $value;
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @param array $items
+     * @param bool|true $override
+     */
+    public function addQueryParameters(array $items, $override = true)
+    {
+        foreach ($items as $key => $value) {
+            $this->addQueryParameter($key, $value, $override);
+        }
+    }
+
+    /**
+     * @param $key
+     */
+    public function removeQueryParameter($key)
+    {
+        unset($this->queryParameters[$key]);
```

Comment:
Do you mind just adding a little check if `array_key_exists($key, $this->queryParameters)`? Just to prevent a possible notice. Afterwards it should be good to merge :)